### ISSUE TITLE
Cannot convert react-redux

### ISSUE THREAD

**Issue by Danilo-Araujo-Silva** [Kotlin/ts2kt#58]
When trying to convert the react-redux typescript definitions to kotlin the following exception ocurrs:

cd DefinatelyTyped/types/react-redux
ts2kt index.d.ts

``` 
ts2kt version: 0.0.14
Converting index.d.ts
ts2kt: "ImportDeclaration" kind unsupported yet here! (index.d.ts:1:1 to 13:32)
ts2kt: "ImportDeclaration" kind unsupported yet here! (index.d.ts:13:32 to 14:32)
ts2kt: "ComputedPropertyName" kind unsupported yet here! (index.d.ts:25:51 to 25:60)
ts2kt: "ComputedPropertyName" kind unsupported yet here! (index.d.ts:25:69 to 25:78)
ts2kt: "NeverKeyword" kind unsupported yet here! (index.d.ts:25:104 to 25:110)
ts2kt: "NeverKeyword" kind unsupported yet here! (index.d.ts:25:104 to 25:110)
ts2kt: "ExpressionStatement" kind unsupported yet here! (index.d.ts:25:114 to 25:115)
ts2kt: "EmptyStatement" kind unsupported yet here! (index.d.ts:25:116 to 25:117)
ts2kt: "ExpressionStatement" kind unsupported yet here! (index.d.ts:26:53 to 26:61)
/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:30833
  function DelimitedRangesSequence$iterator$ObjectLiteral(this$DelimitedRangesSequence) {
                                                         ^

RangeError: Maximum call stack size exceeded
    at new DelimitedRangesSequence$iterator$ObjectLiteral (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:30833:58)
    at DelimitedRangesSequence.iterator (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:30886:12)
    at new TransformingSequence$iterator$ObjectLiteral (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:28035:58)
    at TransformingSequence.iterator (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:28045:12)
    at toCollection_9 (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:19264:22)
    at toMutableList_10 (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:19278:12)
    at toList_10 (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:19275:33)
    at lines (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:31006:12)
    at replaceIndent (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:29495:19)
    at trimIndent (/usr/local/share/.config/yarn/global/node_modules/ts2kt/kotlin.js:29488:12)
```

**Comment by ScottPierce** [comment]
Similar issue with `react-native`:

```
spierce7$ ts2kt DefinitelyTyped-master/types/react-native/globals.d.ts DefinitelyTyped-master/types/react-native/index.d.ts
ts2kt version: 0.0.19
Converting DefinitelyTyped-master/types/react-native/globals.d.ts
Save declarations:
	./globals.kt
Converting DefinitelyTyped-master/types/react-native/index.d.ts
ts2kt: "ComputedPropertyName" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4741:28 to 4741:40)
ts2kt: "ExpressionStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4741:40 to 4741:42)
ts2kt: "ExpressionStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4741:44 to 4741:79)
ts2kt: "EmptyStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4741:81 to 4741:82)
ts2kt: "ComputedPropertyName" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4746:67 to 4746:79)
ts2kt: "ExpressionStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4746:79 to 4746:81)
ts2kt: "ExpressionStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4746:83 to 4746:105)
ts2kt: "EmptyStatement" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:4746:107 to 4746:108)
ts2kt: "ComputedPropertyName" kind unsupported yet here! (DefinitelyTyped-master/types/react-native/index.d.ts:5212:27 to 5212:68)
/usr/local/lib/node_modules/ts2kt/ts2kt.js:4465
  function mapTypeArguments($receiver, typeArguments, declaration) {
                           ^

RangeError: Maximum call stack size exceeded
    at mapTypeArguments (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4465:28)
    at mapTypeReference (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4446:22)
    at mapTypeToUnion_0 (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4328:55)
    at mapType_0 (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4249:12)
    at mapUnionType (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4399:31)
    at mapTypeToUnion_0 (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4317:36)
    at mapType_0 (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4249:12)
    at mapType (/usr/local/lib/node_modules/ts2kt/ts2kt.js:4230:14)
    at TsInterfaceToKt.TsClassifierToKt.translateAccessor_xriwno$_0 (/usr/local/lib/node_modules/ts2kt/ts2kt.js:2702:72)
    at TsInterfaceToKt.TsClassifierToKt.visitIndexSignature_53if9s$ (/usr/local/lib/node_modules/ts2kt/ts2kt.js:2692:10)
```

**Comment by bashor** [comment]
Crash is fixed in 0.0.20 and it will be available on NPM soon

**Comment by ScottPierce** [comment]
@bashor The output seems to be wrong when doing the conversion on react native.

[react-native-result.zip](https://github.com/Kotlin/ts2kt/files/1570320/react-native-result.zip)

React natives several thousand line file is converted to a 37 line kotlin file:
```
@file:Suppress("INTERFACE_WITH_SUPERCLASS", "OVERRIDING_FINAL_MEMBER", "RETURN_TYPE_MISMATCH_ON_OVERRIDE", "CONFLICTING_OVERLOADS", "EXTERNAL_DELEGATION", "NESTED_CLASS_IN_EXTERNAL_INTERFACE")
@file:JsQualifier("global")
package react.native

import kotlin.js.*
import kotlin.js.Json
import org.khronos.webgl.*
import org.w3c.dom.*
import org.w3c.dom.events.*
import org.w3c.dom.parsing.*
import org.w3c.dom.svg.*
import org.w3c.dom.url.*
import org.w3c.fetch.*
import org.w3c.files.*
import org.w3c.notifications.*
import org.w3c.performance.*
import org.w3c.workers.*
import org.w3c.xhr.*

external fun require(name: String): Any = definedExternally
external fun Console.error(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.info(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.log(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.warn(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.trace(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.debug(message: Any? = definedExternally /* null */, vararg optionalParams: Any): Unit = definedExternally
external fun Console.table(vararg data: Any): Unit = definedExternally
external var Console.disableYellowBox: Boolean get() = definedExternally; set(value) = definedExternally
external var Console.ignoredYellowBox: Array<String> get() = definedExternally; set(value) = definedExternally
external var console: Console = definedExternally
external var Navigator.product: String get() = definedExternally; set(value) = definedExternally
external var Navigator.geolocation: GeolocationStatic get() = definedExternally; set(value) = definedExternally
external var navigator: Navigator = definedExternally
external var originalXMLHttpRequest: Any = definedExternally
external var __BUNDLE_START_TIME__: Number = definedExternally
external var ErrorUtils: ErrorUtils = definedExternally
external var __DEV__: Boolean = definedExternally
```
